[Loo Xin Wei] iP

redpelican2108 opened the pull request:
# Narcibot

> "I don't know what I'm supposed to do." -that's what she said
*Narcibot* is short for narcissistic bot, the self-claimed best bot in the world that can help track your tasks. It's features include
- tracking tasks
- insulting you
- praising itself

All you need to do is,

1. download it from [here](https://github.com/redpelican2108/ip).
2. double-click it.
3. add tasks.
4. get ignored by the bot

Features:

- [x] Track Tasks
- [ ] Being useful
- [x] Insult you :wink:

And did you know that it's **Free**?
 
If you'd like to personalize your Narcibot, you can change it with Java. Here's the `main` method.
```java
public static void main(String[] args) {
        new Narcibot("tasks.txt","./data").run();
    }
```

Commit:
Add Gradle support

redpelican2108 committed:
Added Narcibot

redpelican2108 committed:
Added add and list function

redpelican2108 committed:
Added markable tasks

redpelican2108 committed:
Added ToDo, Event, Deadline

redpelican2108 committed:
Added TextUiTesting

redpelican2108 committed:
Handle Errors

redpelican2108 committed:
Added delete

redpelican2108 committed:
Add save feature

redpelican2108 committed:
Add Date recognition

redpelican2108 committed:
Merge branch 'branch-Level-8'

redpelican2108 committed:
Add Parser, Storage, TaskList, Ui Class

redpelican2108 committed:
Move to package

redpelican2108 committed:
Add JUnit

redpelican2108 committed:
Add jar

redpelican2108 committed:
Add JavaDoc

redpelican2108 committed:
Update coding style

redpelican2108 committed:
Add find feature

redpelican2108 committed:
Merge branch 'branch-A-CodingStandard'

redpelican2108 committed:
Merge branch 'branch-Level-9'

redpelican2108 committed:
Add Javadoc for Find

Ardentsoul commented on src/main/java/narcibot/Deadline.java:
```diff
@@ -0,0 +1,74 @@
+package narcibot;
+import java.time.LocalDate;
```
Shouldn't there be a new line after package? 🤔

joey-chance replied:
Yes, I think so too. 🤔

Ardentsoul commented on src/main/java/narcibot/Deadline.java:
```diff
@@ -0,0 +1,74 @@
+     * @param done status of the deadline
+     */
+
+    public Deadline(String name, String time, boolean done) {
```
Shouldn't the boolean parameter have a prefix attached such as is, has, was? 🤔
I noticed this issue in several parts of your code.

joey-chance replied:
I agree with @Ardentsoul, perhaps a more intuitive variable name here?

Ardentsoul commented on src/main/java/narcibot/Event.java:
```diff
@@ -0,0 +1,73 @@
+
+    /**
+     * Constructor for event class
+     * @param name name of the event
```
Shouldn't there be a new line after the description of your javadocs? 🤔
I noticed this in several parts of your code.

joey-chance replied:
Shouldn't there be a new line separating your description and the parameters in your javadocs? 🤔 I noticed the same issue in several other places too.

Ardentsoul commented on src/main/java/narcibot/Narcibot.java:
```diff
@@ -0,0 +1,126 @@
+        }
+        return false;
+    }
+
```
I liked the way your made your code clean and readable. Keep it up! 👍

Ardentsoul reviewed (commented):
LGTM. Just a few nits to fix.

redpelican2108 committed:
Update README.md

redpelican2108 committed:
Update README.md

redpelican2108 committed:
Merge branch 'add-gradle-support' into branch-Level-10

redpelican2108 committed:
Add Gradle

redpelican2108 committed:
Add Java GUI

joey-chance commented on src/main/java/narcibot/Deadline.java:
```diff
@@ -0,0 +1,74 @@
+     */
+    @Override
+    public String getStatus() {
+        return "[D]" + super.getStatus() +" (by: " + time +")";
```
Shouldn't there be a whitespace after the "+" operator here? 🤔 I noticed the same issue in several other places too.

joey-chance commented on src/main/java/narcibot/Narcibot.java:
```diff
@@ -0,0 +1,240 @@
+                break;
+            case "deadline":
+                if (command.length != 3) {
+                    throw new IncorrectFormatException("You want me to remind you of something but you won't tell me of what it is? The format is deadline (task) /by (time)");
```
Shouldn't line length be no longer than 120 characters? 🤔  I have noticed the same issue in several other places too.

joey-chance reviewed (commented):
LGTM 👍 . Some cosmetic changes required.

redpelican2108 committed:
Fix Coding Style

There exists several parts that are yet to adhere with the coding style.

Let's fix that to improve readability

redpelican2108 committed:
Refactor Code

Parts of the code has nested if clauses.

Let's use guard clauses to make the happy path prominent and make the code readable

redpelican2108 committed:
Add assertions

redpelican2108 committed:
Merge pull request #1 from redpelican2108/branch-A-Asssertions

Add assertions

redpelican2108 committed:
Merge branch 'master' into branch-A-CodeQuality

redpelican2108 committed:
Merge pull request #2 from redpelican2108/branch-A-CodeQuality

Refactor Code

redpelican2108 committed:
Add update function

The original task list doesn't allow the changing of deadlines and events without deleting and adding the task again.

Let's add an update function so that it can be done with more ease and it saves more time

redpelican2108 committed:
Add User Guide

The previous version  doesn't contain a user guide. Let's add a user guide.

redpelican2108 committed:
Fix taskList not available error
